### PR TITLE
[sparkle] internal: refactor packaging, compile tsc & tailwind

### PR DESCRIPTION
**Pull request by tdraier**
fixes : [#989](https://github.com/dust-tt/tasks/issues/989)

## Description

Generate esm lib instead of cjs bundle.

In order to use, css need to be imported : 
`import "@dust-tt/sparkle/dist/sparkle.css"`

-> https://github.com/dust-tt/dust/tree/fix/sparkle-packaging-usage

Rollup configurations and dependencies not removed yet.

## Risk

UI glitches and conflicts. Can be rollbacked easily.

## Deploy Plan

Build and deploy sparkle , then use it in front

### CODE REVIEW

File: sparkle/src/icons/index.ts [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+  UserArrow as UserArrowIcon,
+  UserGroup as UserGroupIcon,
+  User as UserIcon,
+  UserPlus as UserPlusIcon,
+  Wrench as WrenchIcon,
+  XCircle as XCircleIcon,
+  XMark as XMarkIcon,
+} from "./solid";
+export {
+  ArrowDownCircle as ArrowDownCircleStrokeIcon,
```
Comment (flvndvd) [inline sparkle/src/icons/index.ts:119]: So we export the same icons into different names?

Comment (tdraier) [reply sparkle/src/icons/index.ts:119]: not in main export, but internally yes. I'll check if these names are used, I don't see the point of keeping both names

File: sparkle/src/_index.ts [hunk 2]
```diff
@@ -1,716 +0,0 @@
-import { SparkleContext } from "./context";
```
Comment (flvndvd) [inline sparkle/src/_index.ts:1]: 😘

**Review by flvndvd — APPROVED** [review]
🚀

**Pull request merged**